src/commands/mod.rs

```diff
@@ -271,6 +272,49 @@ impl EdgeAppManifest {
         write!(&manifest_file, "---\n{yaml}")?;
         Ok(())
     }
+
+    pub fn validate_yaml_manifest(path: &Path) -> Result<(), CommandError> {
+        let data = fs::read_to_string(path).map_err(|_| CommandError::ValidationError("Failed to read file".to_string()))?;
+        let manifest_result: Result<EdgeAppManifest, serde_yaml::Error> = serde_yaml::from_str(&data);
+    
+        match manifest_result {
+            Err(e) => {
+                return Err(CommandError::ValidationError(format!("Failed to deserialize YAML: {}", e)));
+            },
+            Ok(manifest) => {
+                let mut errors = Vec::new();
+    
+                if manifest.app_id.is_empty() {
+                    errors.push("app_id is empty".to_string());
+                }
+    
+                if manifest.settings.is_empty() {
+                    errors.push("settings are empty".to_string());
+                } else {
+                    for (i, setting) in manifest.settings.iter().enumerate() {
```

Comment:
This is not a good approach. I don't like it. Much more natural is to make a custom deserializer that will not accept an empty string.